File: Selenium/Project/Python_Project/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Initialize the WebDriver
driver = webdriver.Chrome()

try:
    # Navigate to the website
    driver.get('https://alchemy.hguy.co/jobs/')
    
    # Locate the navigation menu and click the menu item that says “Post a Job”
    post_job_link = WebDriverWait(driver, 10).until(
        EC.element_to_be_clickable((By.LINK_TEXT, "Post a Job"))
    )
    post_job_link.click()
    
    # Fill in the necessary details
    title = driver.find_element(By.ID,"job_title")
    title.send_keys("Test Job")
    
    description = driver.find_element(By.ID,"job_description_ifr")
    driver.switch_to.frame(description)
    description_body = driver.find_element("xpath","//body[@id='tinymce']")
    description_body.send_keys("This is a test job description.")
    driver.switch_to.default_content()
    
    # Click the button that says “Preview”
    submit_preview = driver.find_element(By.NAME,"continue")
    submit_preview.click()
    
    # Click on the button that says “Submit Listing”
    submit_listing = driver.find_element(By.NAME,"submit_job")
    submit_listing.click()
    
    # Verify that the job listing was posted by visiting the jobs page
    driver.get('https://alchemy.hguy.co/jobs/jobs/')
    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.CLASS_NAME, "job_listing")))
    job_titles = driver.find_elements(By.CSS_SELECTOR,".job_listing .job_listing_title")
    job_titles_text = [job.text for job in job_titles]
    assert "Test Job" in job_titles_text, "Job listing was not posted"
    print("Job listing posted successfully.")

finally:
    # Close the browser
    driver.quit()

File: Selenium/Project/Project_Java/Activity7.java
package Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity7 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
// Navigate to the Alchemy Jobs site
        driver.get("https://alchemy.hguy.co/jobs/");

        // Locate the navigation menu and click the menu item that says “Post a Job”
        WebElement postJobMenuItem = driver.findElement(By.linkText("Post a Job"));
        postJobMenuItem.click();

        // Fill in the necessary details
        WebElement jobTitle = driver.findElement(By.id("job_title"));
        jobTitle.sendKeys("Test Job");

        WebElement jobDescriptionFrame = driver.findElement(By.id("job_description_ifr"));
        driver.switchTo().frame(jobDescriptionFrame);
        WebElement jobDescription = driver.findElement(By.id("tinymce"));
        jobDescription.sendKeys("This is a test job description.");
        driver.switchTo().defaultContent();

        WebElement jobTypeDropdown = driver.findElement(By.id("job_type"));
        Select jobTypeSelect = new Select(jobTypeDropdown);
        jobTypeSelect.selectByVisibleText("Full Time");

        WebElement jobLocation = driver.findElement(By.id("job_location"));
        jobLocation.sendKeys("Remote");

        // Click the button that says “Preview”
        WebElement previewButton = driver.findElement(By.name("continue"));
        previewButton.click();

        // Click on the button that says “Submit Listing”
        WebElement submitListingButton = driver.findElement(By.name("submit_job"));
        submitListingButton.click();

        // Verify that the job listing was posted by visiting the jobs page
        driver.get("https://alchemy.hguy.co/jobs/jobs/");
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(0));
        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.className("job_listing"), "Test Job"));

        if (driver.getPageSource().contains("Test Job")) {
            System.out.println("Job listing was posted successfully.");
        } else {
            System.out.println("Job listing was not posted.");
        }

        // Close the browser
        driver.quit();
    }
}
